frontend/src/features/admin-form/create/payment/PaymentTab.tsx

```diff
@@ -4,18 +4,25 @@ import { useAdminForm } from '~features/admin-form/common/queries'
 
 import { PaymentContent } from './PaymentContent'
 import { PaymentDrawer } from './PaymentDrawer'
+import { defaultPaymentsField } from './usePaymentStore'
 
 export const PaymentTab = (): JSX.Element => {
   const { data: form } = useAdminForm()
 
-  return form?.responseMode === FormResponseMode.Encrypt &&
-    form?.payments_field ? (
+  const isEncryptMode = form?.responseMode === FormResponseMode.Encrypt
+
+  const paymentsField =
+    isEncryptMode && form?.payments_field
+      ? form.payments_field
+      : defaultPaymentsField
```

Comment:
is the reason for needing `defaultPaymentsField` here because currently payments_field is undefined for all forms? If so then i would propose making the `paymentsField` input to `PaymentDrawer` and `PaymentContent` optional, and allowing the children to handle undefined input. For two reasons:

1. Since email mode forms aren't allowed to have payments fields, we should not pass defaults to the children at all, they should be undefined. (If we want to lock the toggle to be disabled in the payment drawer because of email mode, we should hardcode both the "isDisabled" and "value" attributes to guarantee this, rather than relying on the default defined elsewhere in code.) 
2. Basically no additional handling needs to happen for this work - the payment drawer and payment content elements both accept optional input as it is.

What this allows us to do is pass the `form?.payments_field` directly to the children (along with `isEncryptMode`, where necessary `responseMode` checks need to occur), and avoids passing invalid state around (i.e. an email mode form where `paymentsField === { enabled: false }`)

Reply:
In my latest commit [7be108e](https://github.com/opengovsg/FormSG/pull/6011/commits/7be108e9a5abe7f7cad0adadf127b406ab5176b5) I faced an issue when I do not set enabled as a boolean, as FormPaymentsField require enabled to exist. Hence, if I do not set `paymentCommon = {enabled:false}`, I get a bunch of Typescript errors. Let me know if you have a way to circumvent this!

However, in terms of code logic, have followed your suggestions thanks @justynoh! So we do not send a default field to the drawer, and the disabled display is already hardcoded in.

Additionally, have ensured that when payment input is disabled, there will be no mutations

Reply:
Additionally, I think the concept of `payments_field.enabled` is still iffy, and we should use the existence of `payments_channel` as the 'enabling factor' (after #5848 is done), and `enabled` doesn't exist in the design figma either.

Reply:
For your first comment, doing `const { paymentsData: { amount_cents: paymentAmountCents, ...paymentCommon }, ... } = usePaymentStore(...)` in `PaymentDrawer` seems to be fine for me... unless I'm misunderstanding your comment?

Also, not related to your PR but I wonder if we could pick a better variable name than `paymentCommon` for what it's supposed to represent...

For your second comment, you're right - we can revisit it in the future.

Reply:
It will display another error here when you try to open to payment tab: 
<img width="994" alt="image" src="https://user-images.githubusercontent.com/59867455/228742687-b65d07e0-e3be-4c0b-9525-851516421570.png">
As we use zustand store so Typescript will 'assume' FormPaymentsField exists when it doesn't 😅

And encrypt guarding the useEffect in paymentdrawer doesn't work too, honestly I'm having some difficulty understanding how we are using zustand's store

Reply:
as discussed, can make data in the store optional and set the default in `PaymentDrawer`'s useEffect, a la end page store

Reply:
[1585f40](https://github.com/opengovsg/FormSG/pull/6011/commits/1585f409d8df9383d0f3943908bb1f6ce8cd8006)

Have completely removed the need for a default value, following end page store's implementation. As either way the default value seems to almost always get set as undefined by `setToInactive()`

Have make some slight improvements over the two commits as per discussed with @justynoh, thanks so much!!!